src/sentry/seer/anomaly_detection/store_data.py

```diff
@@ -219,24 +220,38 @@ def fetch_historical_data(
         dataset_label = "errors"
     dataset = get_dataset(dataset_label)
 
-    if not project or not dataset or not alert_rule.organization:
+    if not project or not alert_rule.organization:
         return None
 
+    snuba_params = SnubaParams(
+        organization=alert_rule.organization,
+        projects=[project],
+        start=start,
+        end=end,
+    )
+
     if dataset == metrics_performance:
-        return get_crash_free_historical_data(
+        historical_data = get_crash_free_historical_data(
             start, end, project, alert_rule.organization, granularity
         )
-
+    elif "apdex" in snuba_query.aggregate:
+        generator = get_stats_generator(use_discover=True, remove_on_demand=False)
```

Comment:
this is pulled from the `metrics-estimation-endpoint` [here](https://github.com/getsentry/sentry/blob/master/src/sentry/api/endpoints/organization_on_demand_metrics_estimation_stats.py#L79) which then gets called in in `organization_events.py` [here](https://github.com/getsentry/sentry/blob/master/src/sentry/api/bases/organization_events.py#L466). I _think_ this is all that's needed for the type of apdex metric alert you can create from the UI, and all the other stuff is for Discover queries (and I support custom metric type metric alerts?).